lib/commands.ex

```diff
@@ -0,0 +1,163 @@
+defmodule Inky.Commands do
+  defmodule State do
+    @state_fields [:io_mod, :io_state]
+
+    @enforce_keys @state_fields
+    defstruct @state_fields
+  end
+
+  @default_io_mod Inky.RpiIO
+
+  # API
+  def init_io(io_mod \\ @default_io_mod, io_args \\ []) do
+    %State{io_mod: io_mod, io_state: io_mod.init(io_args)}
+  end
```

Comment:
1. I should have named this function "init"
2. it should have taken a keyword list, named opts